### PR TITLE
[plugin.video.101barz] 1.0.0

### PR DESCRIPTION
**Pull request by skipmodea1**
### Description
1.0.0 (2019-02-13)
- initial version
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

File: plugin.video.101barz/addon.xml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<addon
+	id="plugin.video.101barz"
+	name="101barz"
+	version="1.0.0"
+	provider-name="Skipmode A1">
+    <requires>
+        <import addon="xbmc.python"                     version="2.25.0"/>
```
Comment (Rechi) [inline plugin.video.101barz/addon.xml]: > WARN: For leia it is advised to set xbmc.python version to 2.26.0

File: plugin.video.101barz/default.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    add_dir(LANGUAGE(30005), ROTJOCHLOUNGEURL, 'list_videos', "")
+    add_dir(LANGUAGE(30006), HITMAKERZURL, 'list_videos', "")
+    add_dir(LANGUAGE(30007), BARZ4BARZURL, 'list_videos', "")
+    add_dir(LANGUAGE(30008), TALENTVDMAANDURL, 'list_videos', "")
+
+    xbmcplugin.endOfDirectory(PLUGIN_HANDLE)
+
+
+def list_videos(url):
+    log("url", url)
```
Comment (Rechi) [inline plugin.video.101barz/default.py]: full path is already logged by Kodi

File: plugin.video.101barz/default.py [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+        log("search_url", search_url)
+
+        # add search url
+        add_dir(LANGUAGE(30001), search_url, 'search', "")
+
+    html_source = get_url(url)
+    json_data = json.loads(html_source)
+
+    log("json_data", json_data)
```
Comment (Rechi) [inline plugin.video.101barz/default.py]: Is json_data much data?

File: plugin.video.101barz/default.py [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+
+    for item in json_data['video']:
+        log("item", item)
```
Comment (Rechi) [inline plugin.video.101barz/default.py]: already logged above in the json_data log line

File: plugin.video.101barz/default.py [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+
+        length = ""
+        title = item['title']
+        desc = item['subtitle']
+        date = item['publish_up']
+        thumb = item['card_image']
+        youtube_id = item['youtube_id']
+        youtube_url = 'plugin://plugin.video.youtube/play/?video_id=%s' % youtube_id
+
+        log("youtube_url", youtube_url)
```
Comment (Rechi) [inline plugin.video.101barz/default.py]: already logged above in the json_data and item log line

File: plugin.video.101barz/default.py [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+    keyboard.doModal()
+    if keyboard.isConfirmed() and keyboard.getText():
+        search_string = keyboard.getText().replace(" ", "+")
+        search_url = BASE_URL + convertToUnicodeString(url).replace("<searchstring>", search_string)
+        list_videos(search_url)
+
+
+def play_video(video_url):
+
+    log("video_url", video_url)
```
Comment (Rechi) [inline plugin.video.101barz/default.py]: full path is already logged by Kodi

File: plugin.video.101barz/addon.xml (hunk 1 above)
Comment (skipmodea1) [reply plugin.video.101barz/addon.xml]: I set the version to 2.26.0, but now i get 'addon is not compatible with Kodi' without any more info what's wrong with it:
05:05:41.773 T:6584   DEBUG: ------ Window Init (DialogAddonInfo.xml) ------
05:05:41.773 T:6584    INFO: Loading skin file: DialogAddonInfo.xml, load type: KEEP_IN_MEMORY
05:05:41.779 T:6584   DEBUG: CAddonSettings[plugin.video.101barz]: loading setting definitions
05:05:41.779 T:6584   DEBUG: CAddonSettings[plugin.video.101barz]: trying to load setting definitions from old format...
05:05:41.780 T:6584   DEBUG: CAddonSettings[plugin.video.101barz]: loading setting values
05:05:42.806 T:6584    INFO: XCURL::DllLibCurlGlobal::CheckIdle - Closing session to http://mirrors.kodi.tv (easy=000001937376CF30, multi=0000019372E55AD0)
05:05:44.656 T:6584   DEBUG: CInputManager::ProcessMouse: trying mouse action leftclick
05:05:44.657 T:6584   ERROR: Add-on 'plugin.video.101barz' is not compatible with Kodi
05:05:44.659 T:6584   DEBUG: ------ Window Init (DialogNotification.xml) ------
05:05:50.380 T:6584   DEBUG: ------ Window Deinit (DialogNotification.xml) ------
05:05:55.581 T:6584   DEBUG: CInputManager::ProcessMouse: trying mouse action leftclick
05:05:55.581 T:6584   ERROR: Add-on 'plugin.video.101barz' is not compatible with Kodi
05:05:55.584 T:6584   DEBUG: ------ Window Init (DialogNotification.xml) ------
05:06:01.305 T:6584   DEBUG: ------ Window Deinit (DialogNotification.xml) ------
05:06:04.055 T:6584   DEBUG: CInputManager::ProcessMouse: trying mouse action leftclick

Please advise...

Comment (Rechi) [reply plugin.video.101barz/addon.xml]: Are you already using Kodi v18?

Comment (skipmodea1) [reply plugin.video.101barz/addon.xml]: yeah, when starting it says migrating. And then it says the addon is not compatible.

Comment (Rechi) [reply plugin.video.101barz/addon.xml]: Here it works, this looks more like a Kodi issue.

Comment (MartijnKaijser) [reply plugin.video.101barz/addon.xml]: Likely the upgrade didn't detect the bumped version of xbmc.python

File: plugin.video.101barz/default.py [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+        for paramsPair in param_pairs:
+            param_splits = paramsPair.split('=')
+            if (len(param_splits)) == 2:
+                param_dict[param_splits[0]] = param_splits[1]
+    return param_dict
+
+
+def add_link(title, url, mode, iconimage, desc="", duration="", youtube_id = ""):
+    u = sys.argv[0] + "?url=" + urllib.parse.quote_plus(url) + "&mode=" + convertToUnicodeString(mode)
+    liz = xbmcgui.ListItem(title, iconImage="DefaultVideo.png", thumbnailImage=iconimage)
```
Comment (enen92) [inline plugin.video.101barz/default.py]: Deprecated use setart only

File: plugin.video.101barz/default.py [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+    title = title.encode('ascii', 'ignore')
+    # Add refresh option to context menu
+    liz.addContextMenuItems([('Refresh', 'Container.Refresh')])
+    ok = xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=u, listitem=liz)
+    return ok
+
+
+def add_dir(title, url, mode, iconimage, desc=""):
+    u = sys.argv[0] + '?url=' + urllib.parse.quote_plus(url) + '&mode=' + convertToUnicodeString(mode)
+    liz = xbmcgui.ListItem(title, iconImage='DefaultFolder.png', thumbnailImage=iconimage)
```
Comment (enen92) [inline plugin.video.101barz/default.py]: Deprecated use setart only please

File: plugin.video.101barz/default.py [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+        xbmc.log("[ADDON] %s v%s (%s) debug mode, %s = %s" % (
+            ADDON, VERSION, DATE, name_object, "Unable to log the object due to an error while converting it to an unicode string"), xbmc.LOGDEBUG)
+
+
+params = parameters_string_to_dict(sys.argv[2])
+mode = params.get('mode')
+url = params.get('url')
+
+if url is None:
+    if SETTINGS.getSetting('onlyshowallvideoscategory') == 'true':
```
Comment (enen92) [inline plugin.video.101barz/default.py]: I think you now have methods to get the setting as bool

File: plugin.video.101barz/addon.xml (hunk 1 above)
Comment (skipmodea1) [reply plugin.video.101barz/addon.xml]: I found out what was wrong, i was using an older version of kodi 18. It works fine with the latest kodi 18.

File: plugin.video.101barz/default.py (hunk 2 above)
Comment (skipmodea1) [reply plugin.video.101barz/default.py]: I will remove this log.

File: plugin.video.101barz/default.py (hunk 3 above)
Comment (skipmodea1) [reply plugin.video.101barz/default.py]: I will remove this log.

File: plugin.video.101barz/default.py (hunk 4 above)
Comment (skipmodea1) [reply plugin.video.101barz/default.py]: I'm keeping this log as i need it for debugging when something goes wrong in the future.

File: plugin.video.101barz/default.py (hunk 5 above)
Comment (skipmodea1) [reply plugin.video.101barz/default.py]: I will remove this log.

File: plugin.video.101barz/default.py (hunk 6 above)
Comment (skipmodea1) [reply plugin.video.101barz/default.py]: I will remove this log.

File: plugin.video.101barz/default.py (hunk 8 above)
Comment (skipmodea1) [reply plugin.video.101barz/default.py]: Will adjust.

File: plugin.video.101barz/default.py (hunk 7 above)
Comment (skipmodea1) [reply plugin.video.101barz/default.py]: Will adjust.

File: plugin.video.101barz/default.py (hunk 9 above)
Comment (skipmodea1) [reply plugin.video.101barz/default.py]: Will adjust.

**Comment by skipmodea1** [comment]
I think i adjusted the code for most request. Can you look at it again?

**Comment by skipmodea1** [comment]
Bah this Pull request is totally kaput now. I swear updating a PR is like the bane of my existence!

**Pull request closed without merging**